script/gen_gallery.py

```diff
@@ -501,216 +487,200 @@ def generate_rst_files(rst_dir, examples_dir, images_dir):
 
         ebase = ex_filename.replace(examples_dir, '')[1:]
         base_dir = os.path.dirname(ebase)
-
-        rst_filename = os.path.basename(ex_filename).replace('.py', '.rst')
-
+        rst_filename = ebase2fbase(ebase) + '.rst'
         dir_map.setdefault(base_dir, []).append((ex_filename, rst_filename))
 
     for dirname, filenames in six.iteritems(dir_map):
         filenames = sorted(filenames, key=lambda a: a[1])
-        dir_map[dirname ] = filenames
+        dir_map[dirname] = filenames
 
     # Main index.
     mfd = open(os.path.join(rst_dir, 'index.rst'), 'w')
-    mfd.write(_index % ('sfepy', 'Examples', '=' * 8))
+    mfd.write(_index % ('examples', 'Examples', '=' * 8))
 
     for dirname, filenames in ordered_iteritems(dir_map):
-        full_dirname = os.path.join(rst_dir, dirname)
-        ensure_path(full_dirname + os.path.sep)
-
         # Subdirectory index.
-        ifd = open(os.path.join(full_dirname, 'index.rst'), 'w')
-        ifd.write(_index % (dirname, dirname, '=' * len(dirname)))
+        ifd = open(os.path.join(rst_dir, '%s-index.rst' % dirname), 'w')
+        ifd.write(_index % (dirname + '-examples',
+                            dirname, '=' * len(dirname)))
 
         for ex_filename, rst_filename in filenames:
-            full_rst_filename = os.path.join(full_dirname, rst_filename)
-            output('"%s"' % full_rst_filename.replace(rst_dir, '')[1:])
-            rst_filename_ns = rst_filename.replace('.rst', '')
```

Comment:
This is not safe as it replaces all occurrences of '.rst'. Use `os.path.splitext()`, please. (Also at other places in this file.) Or, better, use `edit_filename()` from `sfepy.base.ioutils`.